sklearn/tree/_tree.pyx

```diff
@@ -417,7 +416,7 @@ cdef class BestFirstTreeBuilder(TreeBuilder):
                                     SIZE_t start, SIZE_t end, double impurity,
                                     bint is_first, bint is_left, Node* parent,
                                     SIZE_t depth,
-                                    PriorityHeapRecord* res) nogil:
+                                    PriorityHeapRecord* res) nogil except *:
```

Comment:
-1 is already used as the error return code in the existing API. Let's stay consistent and use `except -1`.